client_integration_test.go

```diff
@@ -635,13 +645,14 @@ func TestClientRename(t *testing.T) {
 	defer cmd.Wait()
 	defer sftp.Close()
 
-	f, err := ioutil.TempFile("", "sftptest")
-	if err != nil {
-		t.Fatal(err)
-	}
+	dir, err := ioutil.TempDir("", "sftptest-rename")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+	f, err := os.Create(filepath.Join(dir, "old"))
+	require.Nil(t, err)
```

Comment:
I only mentioned it on the first line that was relevant, because I don’t want to spam your PR with a whole bunch of linting messages.

However, please sweep the whole file for `require.Nil`, you‘re the only code using `require` and/or `assert` right now (well, and a previous commit from you that I so unhelpfully didn’t vet fully), so it shouldn’t be a whole bunch of change.

A quick look seems like you should be able to a file-wide search-replace of `require.Nil` to `require.NoError`.

Reply:
Sorry, amend/rebase error on my part. I've now caught all instances, I think, including one that wasn't part of this PR to begin with:

```
$ git grep Nil
request-server_test.go: assert.NotNil(t, _foo.ctx)
request-server_test.go: assert.Nil(t, rf)
request-server_test.go: assert.Nil(t, fi)
```

Reply:
Yeah, there were like three from a previous PR that I totally missed during review. :laughing: